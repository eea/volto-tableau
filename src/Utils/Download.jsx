import React from 'react';
import { Popup } from 'semantic-ui-react';
import cx from 'classnames';

const Download = ({ viz }) => {
  const [open, setOpen] = React.useState(false);
  const popupRef = React.useRef();

  return (
    <Popup
      popper={{ id: 'vis-toolbar-popup', className: 'download-popup' }}
      position="bottom left"
      on="click"
      open={open}
      onClose={() => {
        setOpen(false);
      }}
      onOpen={() => {
        setOpen(true);
      }}
      ref={popupRef}
      trigger={
        <div className="tableau-download-container">
          <button className={cx('trigger-button', { open })}>
            <i className="ri-download-fill" />
            Download
          </button>
        </div>
      }
      content={
        <>
          <div className="item">
            <span className="label">Data formats</span>
            <div className="types">
              <div className="type">
                <button
                  onClick={() => {
                    viz.showExportCrossTabDialog();
                    popupRef.current.triggerRef.current.click();
                  }}
                >
                  <span>CSV</span>
                </button>
              </div>
              <div className="type">
                <button
                  onClick={() => {
                    viz.exportCrossTabToExcel();
                    popupRef.current.triggerRef.current.click();
                  }}
                >
                  <span>Excel</span>
                </button>
              </div>
            </div>
          </div>
          <div className="item">
            <span className="label">Image formats</span>
            <div className="types">
              <div className="type">
                <button
                  onClick={() => {
                    viz.showExportImageDialog();
                    popupRef.current.triggerRef.current.click();
                  }}
                >
                  <span>PNG</span>
                </button>
              </div>
            </div>
          </div>
          <div className="item">
            <span className="label">Other formats</span>
            <div className="types">
              <div className="type">
                <button
                  onClick={() => {
                    viz.showExportPDFDialog();
                    popupRef.current.triggerRef.current.click();
                  }}
                >
                  <span>PDF</span>
                </button>
              </div>
            </div>
          </div>
        </>
      }
    />
  );
};

export default Download;
