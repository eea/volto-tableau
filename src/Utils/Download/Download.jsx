import React from 'react';
import { Popup } from 'semantic-ui-react';
import cx from 'classnames';
import { Icon } from '@plone/volto/components';
import downloadSVG from '@eeacms/volto-tableau/icons/download.svg';

const Download = ({ viz }) => {
  const [expanded, setExpanded] = React.useState(false);
  const popupRef = React.useRef();

  return (
    <Popup
      popper={{ id: 'tableau-download-popup' }}
      trigger={
        <div className="tableau-download-container">
          <button className={cx('tableau-download-button', { expanded })}>
            <span>Download</span>
            <Icon name={downloadSVG} size="24px" />
          </button>
        </div>
      }
      position="bottom left"
      on="click"
      onClose={() => {
        setExpanded(false);
      }}
      onOpen={() => {
        setExpanded(true);
      }}
      ref={popupRef}
    >
      <ul className="no-bullets">
        <li>
          Data formats
          <div className="visualization-wrapper">
            <div className="visualization-info">
              <div>
                <button
                  className="tableau-download-button tableau-format-download"
                  onClick={() => {
                    viz.showExportCrossTabDialog();
                    popupRef.current.triggerRef.current.click();
                  }}
                >
                  <span>CSV</span>
                </button>
              </div>
              <div>
                <button
                  className="tableau-download-button tableau-format-download"
                  onClick={() => {
                    viz.exportCrossTabToExcel();
                    popupRef.current.triggerRef.current.click();
                  }}
                >
                  <span>Excel</span>
                </button>
              </div>
            </div>
          </div>
        </li>
        <li>
          Image formats
          <div className="visualization-wrapper">
            <div className="visualization-info">
              <div>
                <button
                  className="tableau-download-button tableau-format-download"
                  onClick={() => {
                    viz.showExportImageDialog();
                    popupRef.current.triggerRef.current.click();
                  }}
                >
                  <span>PNG</span>
                </button>
              </div>
            </div>
          </div>
        </li>
        <li>
          Other formats
          <div className="visualization-wrapper">
            <div className="visualization-info">
              <div>
                <button
                  className="tableau-download-button tableau-format-download"
                  onClick={() => {
                    viz.showExportPDFDialog();
                    popupRef.current.triggerRef.current.click();
                  }}
                >
                  <span>PDF</span>
                </button>
              </div>
            </div>
          </div>
        </li>
      </ul>
    </Popup>
  );
};

export default Download;
