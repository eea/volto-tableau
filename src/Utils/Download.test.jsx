import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-intl-redux';
import configureStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom';

import Download from './Download';

const mockStore = configureStore();

const store = mockStore({
  userSession: { token: '1234' },
  intl: {
    locale: 'en',
    messages: {},
  },
});

describe('Edit', () => {
  it('should render the component', () => {
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Download />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
