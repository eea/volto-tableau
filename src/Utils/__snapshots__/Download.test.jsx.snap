// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit should render the component 1`] = `
<div
  className="tableau-download-container"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <button
    className="trigger-button"
  >
    <i
      className="ri-download-fill"
    />
    Download
  </button>
</div>
`;
